src/util.rs

```diff
@@ -482,3 +484,15 @@ impl FullName for PeripheralInfo {
         self.name.as_str().into()
     }
 }
+
+pub fn group_names(d: &Device) -> Vec<String> {
+    let group_names: Vec<_> = d
+        .peripherals
+        .iter()
+        .filter_map(|p| p.group_name.as_ref())
+        .map(|name| name.to_sanitized_snake_case().to_string())
+        .collect::<HashSet<String>>()
+        .into_iter()
+        .collect();
```

Comment:
What sense of extra allocation?
Just return `HashSet`

Reply:
`HashSet<Cow<str>>`

Reply:
Oh, my mistake. Just want to remove duplicates.